File: frontend/src/api/transfer.ts
import request from './request'

export interface Transfer {
  id: number
  fromAccountId: number
  toAccountId: number
  amount: number
  currency: string
  type: string
  status: string
  createdAt: string
  lastUpdatedAt: string
}

// 发起转账
export function createTransfer(data: {
  fromAccountId: number
  toAccountId: number
  amount: number
  currency: string
  type: string
}) {
  return request.post<any, Transfer>('/transfer/transfers', data)
}

// 查询转账
export function getTransfer(transferId: number) {
  return request.get<any, Transfer>(`/transfer/transfers/${transferId}`)
}

File: transfer-service/src/main/java/com/isbank/transfer/entity/Transfer.java
package com.isbank.transfer.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.isbank.common.enums.TransferStatus;
import com.isbank.common.enums.TransferType;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 转账实体
 */
@Data
@TableName("transfers")
public class Transfer {
    @TableId(type = IdType.AUTO)
    private Long id;
    
    private Long fromAccountId;
    
    private Long toAccountId;
    
    private BigDecimal amount;
    
    private String currency;
    
    private TransferType type;
    
    private TransferStatus status;
    
    private LocalDateTime createdAt;
    
    private LocalDateTime lastUpdatedAt;
}

